test/unit/imageWithPreview.directiveSpec.js

```diff
@@ -35,9 +35,12 @@ describe('ngImageInputWithPreview', function() {
       deferred.resolve(result);
       fileReader.readAsDataUrl = jasmine.createSpy().and.returnValue(deferred.promise);
 
-      file = {
+      file = new Blob([result], {
         type: type,
-      };
+      });
+      file.lastModifiedDate = '';
+      file.name = 'filename';
```

Comment:
I assume the tests failed for you if you didn't include a `name` and the `lastModifiedDate`. In this case, this `Blob` creation would benefit from being extracted into a helper, as it's repeated in quite a few places. E.g. so one could just write `file = createBlob({content: result, type: type});`

Reply:
Indeed. The mock was just assuming a file, rather than actually creating one that we could work with.

More than happy to look at creating a helper, however pushed for time currently so wouldn't plan on doing that in this PR.